charts/aws-fsx-openzfs-csi-driver/templates/controller-serviceaccount.yaml

```diff
@@ -0,0 +1,164 @@
+{{- if .Values.controller.serviceAccount.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Values.controller.serviceAccount.name }}
+  labels:
+    {{- include "aws-fsx-openzfs-csi-driver.labels" . | nindent 4 }}
+  {{- with .Values.controller.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+{{- else }}
+  {{- $exists := (lookup "v1" "ServiceAccount" .Release.Namespace .Values.controller.serviceAccount.name) }}
+  {{- if not $exists }}
+    {{- fail (printf "create serviceaccount %s/%s or set .controller.serviceaccount.create true" .Release.Namespace .Values.controller.serviceAccount.name) }}
+  {{- end }}
+{{- end }}
+
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: fsx-openzfs-csi-external-provisioner-role
+  labels:
+    {{- include "aws-fsx-openzfs-csi-driver.labels" . | nindent 4 }}
+rules:
+  # The following rule should be uncommented for plugins that require secrets
```

Comment:
Do we need to leave this in? If not we can probably remove it from the Kubernetes deploy files as well.

Reply:
Good question. I think there is an avenue for users to configure secrets for themselves if they want to, and leaving this in allows them to do that. I don't feel super strongly about this and I'll need to do more research into the use case for these secrets. I'll need to make a rev to this PR anyways to make the helm chart templates consistent with the deployment manifests, so I'll delete these lines if I see that it's safe to do so.

Reply:
Per conversation with @jacobwolfaws, we should probably leave this in.